tools/topology/topology1/sof-adl-es83x6-ssp1-hdmi-ssp02.m4

```diff
@@ -0,0 +1,256 @@
+# Buffers use s32le format, 1000us deadline with priority 0 on core 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	3, SSP, 0, SSP0-HDMI,
+	PIPELINE_SOURCE_3, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
+
+# capture DAI is SSP0 using 2 periods
+# Buffers use s32le format, 1000us deadline with priority 0 on core 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	4, SSP, 0, SSP0-HDMI,
+	PIPELINE_SINK_4, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
+
+# playback DAI is SSP2 using 2 periods
+# Buffers use s32le format, 1000us deadline with priority 0 on core 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	8, SSP, 2, SSP2-HDMI,
+	PIPELINE_SOURCE_8, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
+
+# capture DAI is SSP2 using 2 periods
+# Buffers use s32le format, 1000us deadline with priority 0 on core 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	9, SSP, 2, SSP2-HDMI,
+	PIPELINE_SINK_9, 2, s32le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
+
+
+# playback DAI is iDisp1 using 2 periods
+# Buffers use s32le format, 1000us deadline with priority 0 on core 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+        5, HDA, 3, iDisp1,
+        PIPELINE_SOURCE_5, 2, s32le,
+        1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp2 using 2 periods
+# Buffers use s32le format, 1000us deadline with priority 0 on core 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+        6, HDA, 4, iDisp2,
+        PIPELINE_SOURCE_6, 2, s32le,
+        1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp3 using 2 periods
+# Buffers use s32le format, 1000us deadline with priority 0 on core 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+        7, HDA, 5, iDisp3,
+        PIPELINE_SOURCE_7, 2, s32le,
+        1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+PCM_DUPLEX_ADD(ES8336, 0, PIPELINE_PCM_1, PIPELINE_PCM_2)
+PCM_DUPLEX_ADD(HDMI-1-In, 1, PIPELINE_PCM_3, PIPELINE_PCM_4)
+PCM_DUPLEX_ADD(HDMI-2-In, 4, PIPELINE_PCM_8, PIPELINE_PCM_9)
+
+PCM_PLAYBACK_ADD(HDMI 1, 5, PIPELINE_PCM_5)
+PCM_PLAYBACK_ADD(HDMI 2, 6, PIPELINE_PCM_6)
+PCM_PLAYBACK_ADD(HDMI 3, 7, PIPELINE_PCM_7)
+
+# BE configurations - overrides config in ACPI if present
+
+#SSP 1 (ID: 0)
+DAI_CONFIG(SSP, SSP_NUM, 0, `SSP'SSP_NUM`-Codec',
+	SSP_CONFIG(I2S, SSP_CLOCK(mclk, SSP_MCLK, codec_mclk_in),
+		SSP_CLOCK(bclk, 2400000, codec_consumer),
+		SSP_CLOCK(fsync, 48000, codec_consumer),
+		SSP_TDM(2, 25, 3, 3),
+		SSP_CONFIG_DATA(SSP, SSP_NUM, 24, 0)))
+
+
+#SSP 0 (ID: 6)
+DAI_CONFIG(SSP, 0, 6, SSP0-HDMI,
+	   SSP_CONFIG(I2S, SSP_CLOCK(mclk, 24576000, codec_mclk_in),
+		      SSP_CLOCK(bclk, 3072000, codec_provider),
+		      SSP_CLOCK(fsync, 48000, codec_provider),
+		      SSP_TDM(2, 32, 3, 3),
+		      SSP_CONFIG_DATA(SSP, 0, 32, 0)))
+
+#SSP 2 (ID: 7)
+DAI_CONFIG(SSP, 2, 7, SSP2-HDMI,
+	   SSP_CONFIG(I2S, SSP_CLOCK(mclk, 24576000, codec_mclk_in),
+		      SSP_CLOCK(bclk, 3072000, codec_provider),
+		      SSP_CLOCK(fsync, 48000, codec_provider),
+		      SSP_TDM(2, 32, 3, 3),
+		      SSP_CONFIG_DATA(SSP, 2, 32, 0)))
```

Comment:
line 230 to here go to hdmi-in.m4

You will need to add defines for SSP0 and SSP2 so that there's no conflict should the primary SSP_NUM change.